docs/concepts/recursion.rst

```diff
@@ -0,0 +1,121 @@
+======================
+Recursive Applications
+======================
+
+Burr supports sub-applications. E.G. running Burr within Burr.
+Currently this capability is only done with tracking (linking applications),
+but we are actively working on building first-level constructs that allow this.
+
+This can enable the following use-cases (among many others):
+
+1. Parallel results to different LLMs for comparison
+2. Multi simultaneous tasks/chains for an LLM
+3. Black-boxing an action as a subset of tasks
+4. Running multiple queries simultaneously and getting an LLM to synthesize.
```

Comment:
isn't this the same as (2) but more simply worded?

Reply:
No on is different questions and the other is the same question but muliple queries